client/data/emails/use-emails-query.js

```diff
@@ -0,0 +1,29 @@
+/**
+ * External dependencies
+ */
+import { useQuery } from 'react-query';
+
+/**
+ * Internal dependencies
+ */
+import wpcom from 'calypso/lib/wp';
+
+const getCacheKey = ( siteId, domain ) => [ 'emails', siteId, domain ];
+
+/**
+ * Get the associated emails given a Site Identificator
+ * and a domain string (example.com)
+ *
+ * @param {number} siteId Site identificator
+ * @param {string} domain Domain
+ * @returns {data, error, isLoading} Returns and object with the
+ * data associated to the SiteId & Domain
+ */
+export const useEmailsQuery = ( siteId, domain ) => {
```

Comment:
Shouldn't we name this `useEmailAccountsQuery()` to better match the type of data returned?